src/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(QUICK_CUDA_CUDA_SOURCE gpu.cu gpu_type.cu gpu_get2e.cu)
+     
+set(QUICK_CUDA_FORTRAN_SOURCE gpu_write_info.f90)
+
+set_property(SOURCE ${QUICK_CUDA_FORTRAN_SOURCE} PROPERTY COMPILE_OPTIONS ${OPT_FFLAGS})
+
+#sadly FindCUDA doesn't support OBJECT libraries, so we have to use a static cuda_add_library
+cuda_add_library(quick_cuda STATIC ${QUICK_CUDA_FORTRAN_SOURCE} ${QUICK_CUDA_CUDA_SOURCE} 
+	OPTIONS ${CUDA_DEVICE_CODE_FLAGS} -DCUDA) 
+target_compile_options(quick_cuda PRIVATE -DCUDA)
+target_link_libraries(quick_cuda xc_gpu xc_cuda)
+make_pic_if_needed(quick_cuda)
```

Comment:
What's this function `make_pic_if_needed`?

Reply:
I made it for when you are building a static library but are going to be linking it into a shared library later.  It makes the static library position independent (passes -fPIC) if CMake is set to build shared libraries.  But actually, hmm, it doesn't really apply here because this project is set up to always build shared libraries regardless of BUILD_SHARED_LIBS